File: api_nestjs_distributed_order_platform/src/presentation/http/mappers/stock.mapper.ts
import { StockEntity } from "src/domain/entities/stock.entity";

export function toStockResponse(stock: StockEntity) {
    return {
        productId: stock.getProductId(),
        availableQuantity: stock.getAvailableQuantity(),
        reservedQuantity: stock.getReservedQuantity(),
        version: stock.getVersion(),
    };
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/presentation/http/mapper/StockMapper.java
package com.project.order.presentation.http.mapper;

import com.project.order.domain.model.Stock;
import com.project.order.presentation.http.dto.StockDto.StockResponse;

public class StockMapper {

    private StockMapper() {}

    public static StockResponse toResponse(Stock stock) {
        return new StockResponse(
                stock.getProductId(),
                stock.getAvailableQuantity(),
                stock.getReservedQuantity(),
                stock.getVersion()
        );
    }
}
